src/transport/AdminPairingTable.h

```diff
@@ -81,12 +93,64 @@ class DLL_EXPORT AdminPairingInfo
         mAdmin  = kUndefinedAdminId;
     }
 
+    CHIP_ERROR StoreUsingKVSMgr(DeviceLayer::PersistedStorage::KeyValueStoreManager & mgr)
+    {
+        char key[KeySize()];
+        ReturnErrorOnFailure(GenerateKey(mAdmin, key, sizeof(key)));
+
+        StorableAdminPairingInfo info;
+        info.mNodeId = Encoding::LittleEndian::HostSwap64(mNodeId);
+        info.mAdmin  = Encoding::LittleEndian::HostSwap16(mAdmin);
+
+        return mgr.Put(key, info);
+    }
+
+    CHIP_ERROR FetchUsingKVSMgr(DeviceLayer::PersistedStorage::KeyValueStoreManager & mgr)
+    {
+        char key[KeySize()];
+        ReturnErrorOnFailure(GenerateKey(mAdmin, key, sizeof(key)));
+
+        StorableAdminPairingInfo info;
+
+        ReturnErrorOnFailure(mgr.Get(key, &info, sizeof(info)));
+        mNodeId    = Encoding::LittleEndian::HostSwap64(info.mNodeId);
+        AdminId id = Encoding::LittleEndian::HostSwap16(info.mAdmin);
+        ReturnErrorCodeIf(mAdmin != id, CHIP_ERROR_INCORRECT_STATE);
+
+        return CHIP_NO_ERROR;
+    }
+
+    static CHIP_ERROR DeleteUsingKVSMgr(DeviceLayer::PersistedStorage::KeyValueStoreManager & mgr, AdminId id)
+    {
+        char key[KeySize()];
+        ReturnErrorOnFailure(GenerateKey(id, key, sizeof(key)));
+
+        return mgr.Delete(key);
+    }
+
 private:
     AdminId mAdmin = kUndefinedAdminId;
     NodeId mNodeId = kUndefinedNodeId;
 
     OperationalCredentials mOpCred;
     AccessControlList mACL;
+
+    static constexpr size_t KeySize() { return sizeof(kAdminTableKeyPrefix) + 2 * sizeof(AdminId); }
+
+    static CHIP_ERROR GenerateKey(AdminId id, char * key, size_t len)
+    {
+        VerifyOrReturnError(len >= KeySize(), CHIP_ERROR_INVALID_ARGUMENT);
+        int keySize = snprintf(key, len, "%s%x", kAdminTableKeyPrefix, id);
+        VerifyOrReturnError(keySize > 0, CHIP_ERROR_INTERNAL);
+        VerifyOrReturnError(len > (size_t) keySize, CHIP_ERROR_INTERNAL);
+        return CHIP_NO_ERROR;
+    }
+
+    struct StorableAdminPairingInfo
+    {
+        uint16_t mAdmin;  /* This field is serialized in LittleEndian byte order */
+        uint64_t mNodeId; /* This field is serialized in LittleEndian byte order */
```

Comment:
Is there a reason these need endianness managed?  I don't see how this serialized data would ever be shared with other nodes.

Reply:
Could the node be running on a system which backs up the data to some central storage, where other entities can fetch and use that data? I am not certain if a certain endianness is really needed for this. It just makes it easier to share the data if its ever needed.

Reply:
Alright, fair enough.  Yes, the storage abstraction doesn't necessarily require that storage stay local to the node, and if it's not local (or if it's mirrored somewhere), standardization to some endianness makes sense.